openhtf/io/output/json_factory.py

```diff
@@ -30,33 +26,19 @@ class OutputToJSON(JSONEncoder):
   """
 
   def __init__(self, filename_pattern=None, inline_attachments=True, **kwargs):
-    super(OutputToJSON, self).__init__(**kwargs)
-    self.filename_pattern = filename_pattern
+    super(OutputToJSON, self).__init__(filename_pattern)
     self.inline_attachments = inline_attachments
-
-  def default(self, obj):
-    if isinstance(obj, BaseException):
-      # Just repr exceptions.
-      return repr(obj)
-    return super(OutputToJSON, self).default(obj)
-
-  # pylint: disable=invalid-name
-  def __call__(self, test_record):
-    assert self.filename_pattern, 'filename_pattern required'
+    self.json_encoder = JSONEncoder(**kwargs)
+ 
+  def serialize_test_record(self, test_record):
+    return self.json_encoder.encode(self.ConvertToDict(test_record))
+  
+  def ConvertToDict(self, test_record):
```

Comment:
`def convert_to_dict()`, we're in a weird spot right now for convention, switching to underscores instead of caps, but let's make the whole file consistent at least (lowercase probably preferable, we're going to do a sweep soon and change over to all lowercase, i think).